pact/Pact/Core/IR/Eval/CoreBuiltin.hs

```diff
@@ -924,29 +956,38 @@ coreReadString :: (CEKEval step b i m, MonadEval b i m) => NativeFunction step b
 coreReadString info b cont handler _env = \case
   [VString s] -> do
     viewEvalEnv eeMsgBody >>= \case
-      PObject envData ->
+      PObject envData -> do
+        chargeGasArgs info $ GObjOp $ ObjOpLookup s $ M.size envData
         case M.lookup (Field s) envData of
           Just (PString p) -> returnCEKValue cont handler (VString p)
           _ -> returnCEK cont handler (VError "read-string failure" info)
       _ -> returnCEK cont handler (VError "read-string failure" info)
   args -> argsError info b args
 
-readKeyset' :: (MonadEval b i m) => T.Text -> m (Maybe KeySet)
-readKeyset' ksn = do
+readKeyset' :: (MonadEval b i m) => i -> T.Text -> m (Maybe KeySet)
+readKeyset' info ksn = do
   viewEvalEnv eeMsgBody >>= \case
-    PObject envData ->
+    PObject envData -> do
+      chargeGasArgs info $ GObjOp $ ObjOpLookup ksn $ M.size envData
       case M.lookup (Field ksn) envData of
         Just (PGuard (GKeyset ks)) -> pure (Just ks)
-        Just (PObject dat) -> parseObj dat
+        Just (PObject dat) -> do
+          chargeGasArgs info $ GObjOp $ ObjOpLookup "keys" objSize
```

Comment:
Ditto here: we're charging gas too often I think.

Reply:
This is also resolved offline I guess, leaving as is and resolving.